### DISCUSSION TITLE
Disappearing First Table when Adding Second Table

### DISCUSSION THREAD

**Discussion by aidrecabrera** [QuestPDF/QuestPDF#672] [Q&A]
## Issue: Disappearing First Table when Adding Second Table

**Description:** I'm currently working on a C# application that uses QuestPDF to generate PDF documents. In my code, I have a first table inside a page, and I want to add a second table after the first table in the same page. However, when I add the second table, the first table seems to disappear in the generated PDF. I want to keep both tables visible on the same page.

Here's how I structured my code:

```csharp
var document = Document.Create(container =>
{
    container.Page(page =>
    {
        // ... Page setup and styling ...

        page.Content().Table(table =>
        {
            // First table content ...
        });

        // Adding a second table, but it causes the first table to disappear
        page.Content().Table(newTable =>
        {
            // Second table content ...
        });
    });
});
```
My full code if it helps:
```csharp
using QuestPDF.Fluent;
using QuestPDF.Helpers;
using QuestPDF.Infrastructure;
using QuestPDF.Previewer;

namespace HTMLToPDFWithQuestPDF
{
    class Program
    {
        static void Main(string[] args)
        {
            var document = Document.Create(container =>
            {
            container.Page(page =>
            {
            page.Size(PageSizes.Letter);
            page.Margin(1, Unit.Centimetre);
            page.PageColor(Colors.White);
            page.DefaultTextStyle(x => x.FontSize(10));
            page.DefaultTextStyle(x => x.FontFamily(Fonts.Calibri));

            page.Header()
                .AlignRight()
                .Text("Appendix 1")
                .FontFamily("Arial")
                .FontSize(10)
                .LineHeight(1.5f)
                .FontColor(Colors.Green.Lighten1);

            page.Content().Table(table =>
            {
                table.ColumnsDefinition(columns =>
                {
                    columns.RelativeColumn();
                    columns.RelativeColumn();
                    columns.RelativeColumn();
                    columns.RelativeColumn();
                });

                // by using custom 'Element' method, we can reuse visual configuration
                table.Cell()
                .ColumnSpan(4)
                .Row(1)
                .Column(1)
                .Element(Block)
                .Text("PURCHASE ORDER")
                .Bold()
                .FontSize(20);

                table.Cell().ColumnSpan(2).Row(2).Column(1).Element(Block);
                table.Cell()
                .ColumnSpan(4)
                .Row(2)
                .Column(1)
                .Padding(5)
                .Text(
                    "Barangay: " +
                    "\n" +
                    "Tel. No:"
                    )
                .LineHeight(1f)
                .FontSize(10);

                table.Cell().ColumnSpan(2).Row(2).Column(3).Element(Block);
                table.Cell()
                .ColumnSpan(2)
                .Row(2)
                .Column(3)
                .Padding(5)
                .Text(
                    "City Municipality: " +
                    "\n" +
                    "Province: "
                    )
                .LineHeight(1f)
                .FontSize(10);

                table.Cell().ColumnSpan(2).Row(3).Column(1).Element(Block);
                table.Cell()
                .ColumnSpan(2)
                .Row(3)
                .Column(1)
                .Padding(5)
                .Text(
                    "Supplier: " +
                    "\n" +
                    "Address: "
                    )
                .LineHeight(1f)
                .FontSize(10);

                table.Cell().ColumnSpan(2).Row(3).Column(3).Element(Block);
                table.Cell()
                .ColumnSpan(2)
                .Row(3)
                .Column(3)
                .Padding(5)
                .Text(
                    "P.O. No.: " +
                    "\n" +
                    "Date: "
                    )
                .LineHeight(1f)
                .FontSize(10);

                table.Cell().ColumnSpan(2).Row(4).Column(1).Element(Block);
                table.Cell().ColumnSpan(2).Row(4).Column(1)
                .Padding(5)
                .Text(
                    "TIN:")
                .FontSize(10);

                table.Cell().ColumnSpan(2).Row(4).Column(3).Element(Block);
                table.Cell().ColumnSpan(2).Row(4).Column(3)
                .Padding(5)
                .Text(
                    "Mode of Procurement:" +
                    "\n" +
                    "Bidding Negotiated Over the Counter")
                .FontSize(10);

                table.Cell().ColumnSpan(4).Row(5).Column(1).Element(Block);
                table.Cell()
               .ColumnSpan(4)
               .Row(5)
               .Column(1)
               .Padding(5)
               .Text(
                    "Gentlemen:" +
                    "\n" +
                    "           Please deliver to this Office the following articles subject to the terms and conditions contained herein.")
               .FontSize(10);

                table.Cell().ColumnSpan(2).Row(6).Column(1).Element(Block);
                table.Cell().ColumnSpan(2).Row(6).Column(1)
                .Padding(5)
                .Text(
                    "Place of Delivery: " +
                    "\n" +
                    "Date of Delivery: "
                    )
                .LineHeight(1)
                .FontSize(10);

                table.Cell().ColumnSpan(2).Row(6).Column(3).Element(Block);
                table.Cell().ColumnSpan(2).Row(6).Column(3)
                .Padding(5)
                .Text(
                    "Delivery Term: " +
                    "\n" +
                    "Payment Term: "
                    )
                .LineHeight(1)
                .FontSize(10);

            });

            // Second table (added after the first table)
            page.Content().Table(newTable =>
            {
                newTable.ColumnsDefinition(columns =>
                {
                    columns.RelativeColumn();
                    columns.RelativeColumn();
                    columns.RelativeColumn();
                });

                newTable.Cell()
                    .ColumnSpan(3)
                    .Row(1)
                    .Column(1)
                    .Element(Block)
                    .Text("Second Table Content")
                    .Bold()
                    .FontSize(16);
            });

            page.Footer()
            .AlignCenter()
            .Text(x =>
            {
                x.Span("Page ");
                x.CurrentPageNumber();
            });

            });
            });

            document.ShowInPreviewer();

        }
        static IContainer Block(IContainer container)
        {
            return container
                .Border(0.5f)
                .ShowOnce()
                .MinWidth(50)
                .MinHeight(25)
                .AlignCenter()
                .AlignMiddle();
        }
    }
}
```

**Issue:** After adding the second table, the first table no longer appears in the generated PDF. I would like to know how I can keep both tables visible on the same page.

PDF before adding the second table:
![image](https://github.com/QuestPDF/QuestPDF/assets/61798731/cd9c5637-2d52-4dcc-b2a0-5971337a8a2c)
PDF after adding the second table:
![image](https://github.com/QuestPDF/QuestPDF/assets/61798731/3e42d4cd-e2e9-4521-b47f-e9785cee666c)

**Questions:**

1. What is the reason for the first table disappearing when I add the second table?
2. How can I modify my code to ensure that both tables are displayed on the same page in the generated PDF?

Any help or guidance on how to achieve this would be greatly appreciated!

**Comment by bennetbo** [comment]
You are overriding the `Content()` element, this is why youre first table disappears after adding a second one.
Essentially you are doing this...
```csharp
page.Content = //some content
page.Content = // some other content, overriding the first content
```

The correct way to do it is to use some kind of container element, e.g. Column
```
page.Content().Column(c =>
{
  c.Item().Table(...) // first table
  c.Item().Table(...) // second table
});
```

**TLDR**: Content() can only be assigned once

Sorry if the formatting is messed up, i wrote this on mobile.

**Reply by aidrecabrera** [reply]
I like your response it is concise and very informative. Thank you so much! It's working now.

![image](https://github.com/QuestPDF/QuestPDF/assets/61798731/0db72806-7e61-41b9-b4d9-599c2116bd90)

Have a great day.
